firefox_puppeteer/tests/test_toolbars.py

```diff
@@ -0,0 +1,132 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette import Wait
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+from firefox_ui_harness.decorators import skip_under_xvfb
+
+
+class TestLocationBar(FirefoxTestCase):
+
+    def test_reload(self):
+        event_types = ["shortcut", "shortcut2", "button"]
+        for event in event_types:
+            # TODO: Until we have waitForPageLoad, this only tests API
+            # compatibility.
+            self.browser.navbar.locationbar.reload_url(event, True)
+            self.browser.navbar.locationbar.reload_url(event, False)
+
+    def test_focus_and_clear(self):
+        locationbar = self.browser.navbar.locationbar
+        locationbar.urlbar.send_keys("zyx")
+        locationbar.clear()
+        self.assertEqual(locationbar.value, '')
+        locationbar.urlbar.send_keys("zyx")
+        self.assertEqual(locationbar.value, 'zyx')
+        locationbar.clear()
+        self.assertEqual(locationbar.value, '')
+
+    def test_load_url(self):
+        data_uri = 'data:text/html,<title>Title</title>'
+        locationbar = self.browser.navbar.locationbar
+        locationbar.clear()
+        locationbar.load_url(data_uri)
```

Comment:
I think that load_url() should clear the location bar value itself. That is something we never fixed in our Mozmill module.